pkg/operator/ceph/nfs/nfs.go

```diff
@@ -276,6 +300,14 @@ func validateGanesha(context *clusterd.Context, clusterInfo *cephclient.ClusterI
 	// The existence of the pool provided in n.Spec.RADOS.Pool is necessary otherwise addRADOSConfigFile() will fail
 	_, err := cephclient.GetPoolDetails(context, clusterInfo, n.Spec.RADOS.Pool)
```

Comment:
How about splitting out the call to get the pool details and create the default pool into a new method? Then call it immediately after `validateGanesha()`. It's just awkward to create the pool as part of a validation method.

Reply:
Then it will also be simpler to unit test the new method when separated. Please look at adding a test to controller_test.go. There are examples in that file about how to mock the ceph commands.